File: library/src/main/java/io/ashdavies/rx/tasks/SingleEmitterFailureListener.java
package io.ashdavies.rx.tasks;

import android.support.annotation.NonNull;
import com.google.android.gms.tasks.OnFailureListener;
import io.reactivex.SingleEmitter;

class SingleEmitterFailureListener<T> implements OnFailureListener {

  private final SingleEmitter<T> emitter;

  SingleEmitterFailureListener(SingleEmitter<T> emitter) {
    this.emitter = emitter;
  }

  @Override
  public void onFailure(@NonNull Exception exception) {
    emitter.onError(exception);
  }
}

File: src/main/kotlin/io/ashdavies/rx/rxtasks/SingleEmitterFailureListener.kt
package io.ashdavies.rx.rxtasks

import com.google.android.gms.tasks.OnFailureListener
import io.reactivex.SingleEmitter

internal class SingleEmitterFailureListener<T>(private val emitter: SingleEmitter<T>) : OnFailureListener {

  override fun onFailure(exception: Exception) = emitter.onError(exception)
}
